build-scripts/build_windows_installer.bat

```diff
@@ -45,6 +45,32 @@ if defined VS160COMNTOOLS (
 if defined VS150COMNTOOLS (

```

Comment:
You also need to add:

```
if defined VS170COMNTOOLS(
  set TOOL_CHAIN_TAG=VS2022
)
```

This detects if the user is building from inside a "x64 Native Tools Command Prompt for VS 2022" and overrides all other compiler version detection.